pkg/api/pod/util_test.go

```diff
@@ -579,6 +579,113 @@ func TestDropAlphaVolumeDevices(t *testing.T) {
 	}
 }
 
+func TestDropFSGroupFields(t *testing.T) {
+	nofsGroupPod := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "container1",
+						Image: "testimage",
+					},
+				},
+			},
+		}
+	}
+
+	var podFSGroup int64 = 100
+	changePolicy := api.AlwaysChangeVolumePermission
+
+	fsGroupPod := func() *api.Pod {
+		return &api.Pod{
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "container1",
+						Image: "testimage",
+					},
+				},
+				SecurityContext: &api.PodSecurityContext{
+					FSGroup:             &podFSGroup,
+					FSGroupChangePolicy: &changePolicy,
+				},
+			},
+		}
+	}
+	podInfos := []struct {
+		description                  string
+		featureEnabled               bool
+		newPodHasFSGroupChangePolicy bool
+		pod                          func() *api.Pod
```

Comment:
IMO it'd be much more flexible if we had `newPod` and `oldPod` instead. Then we could use the builder pattern to create custom pods (with different FSGroups and policies). The for loop below would be cleaner as well.

Since this is API, it's probably worth it to test that many combinations of old and new pods.

Reply:
so if I understand this correctly, you are saying that test should supply separate newpod and oldpod with some basic attributes and then we can use builder pattern to add stuff to pod as test case demands something? 

It is probably worth doing but I do not think it will result in any cleaner code in this case. These test cases aren't building pod object progressively.

Reply:
Not strong opinions about this one. Would you consider this a blocker issue for PR? I am happy to fix this in a follow up.

Reply:
Follow-up is fine